src/metagraph.jl

```diff
@@ -1,91 +1,101 @@
 """
-    MetaGraph{T,Label,Graph,VertexMeta,EdgeMeta,GraphMeta,WeightFunction,U} <: AbstractGraph{T}
+    MetaGraph{T<:Integer,Label,Graph,VertexMeta,EdgeMeta,GraphMeta,WeightFunction,U<:Real} <: AbstractGraph{T}
 
 A graph type with custom vertex labels containing vertex-, edge- and graph-level metadata.
 
 Vertex labels have type `Label`, while vertex (resp. edge, resp. graph) metadata has type `VertexMeta` (resp. `EdgeMeta`, resp. `GraphMeta`).
-It is recommended not to set `Label` to `Int` to avoid confusion between vertex labels and vertex codes (which have type `T <: Integer`).
+It is recommended not to set `Label` to an integer type, so as to avoid confusion between vertex labels and vertex codes (which have type `T<:Integer`).
 
 # Fields
 - `g::Graph`: underlying, data-less graph with vertex indices of type `T`
-- `vprops::Dict{Label,Tuple{T,VertexMeta}}`: dictionary mapping vertex labels to vertex codes and metadata
+- `labels::Dict{T,Label}`: dictionary mapping vertex codes to vertex labels
+- `vcodes::Dict{Label,T}`: dictionary mapping vertex labels to vertex codes
+- `vprops::Dict{Label,VertexMeta}`: dictionary mapping vertex labels to vertex metadata
 - `eprops::Dict{Tuple{Label,Label},EdgeMeta}`: dictionary mapping edge labels such as `(label_u, label_v)` to edge metadata
 - `gprops::GraphMeta`: graph metadata
 - `weightfunction::WeightFunction`: function defining edge weight from edge metadata
 - `defaultweight::U`: default weight for the edges
-- `metaindex::Dict{T,Label}`: dictionary mapping vertex codes to vertex labels
 """
 struct MetaGraph{
-    T<:Integer,
-    Label,
-    Graph,
-    VertexMeta,
-    EdgeMeta,
-    GraphMeta,
-    WeightFunction,
-    U<:Real,
+    T<:Integer,Label,Graph,VertexMeta,EdgeMeta,GraphMeta,WeightFunction,U<:Real
 } <: AbstractGraph{T}
     graph::Graph
-    vprops::Dict{Label,Tuple{T,VertexMeta}}
+    labels::Dict{T,Label}
```

Comment:
I was never much of a fan of these fieldnames. I kept them to match with the original MetaGraphs. I think abbreviations make code hard to read, so I would have chosen something like `vertex_codes` and `vertex_properties`. If we are changing the fieldnames, though, we might want to release a new minor version, because we documented them and someone might be relying on them.

Reply:
I agree with you on this. In fact, that is the convention I chose for my own personal twist on MetaGraphsNext, [DataGraphs](https://github.com/gdalle/DataGraphs.jl). I'll change it here then.

In any case, I think a PR of this size warrants a minor version release

Reply:
Oh, maybe I misread. Why are you splitting up the `label -> (code, metadata)` Dict? Seems like that could lead to extra allocations?